engine/servermaster/campaign.go

```diff
@@ -47,16 +48,14 @@ func (s *Server) generateSessionConfig() (*cluster.EtcdSessionConfig, error) {
 
 func (s *Server) leaderLoop(ctx context.Context) error {
 	var (
-		leaderResignFn context.CancelFunc
-
 		retryInterval    = time.Millisecond * 200
 		needResetSession = false
+
+		// After a leader is resigned, the server will stop to campaign itself
+		// for 10 seconds to avoid the server becoming a leader again.
+		stopCampaignInterval = time.Second * 10
```

Comment:
Is there a particular reason to choose 10s?

Reply:
No 😂. Just to avoid it becoming the leader again.